commons/helpers/pods_helper.py

```diff
@@ -137,29 +137,52 @@ def get_all_pods_containers(self, pod_prefix, pod_list=None):
 
         return pod_containers
 
-    def create_pod_replicas(self, num_replica, deploy=None, pod_name=None):
+    def create_pod_replicas(self, num_replica, deploy=None, pod_name=None, set_name=None):
         """
         Helper function to delete/remove/create pod by changing number of replicas
         :param num_replica: Number of replicas to be scaled
         :param deploy: Name of the deployment of pod
-        :param pod_name: Name of the pod
-        :return: Bool, string (status, deployment name)
+        :param pod_name: Name of the pod (Compulsory param for Statefulset pods, User needs to
+        mention last pod of the Statfulset if set_type is Statefulset)
+        :param set_name: Name of the set (i.e Statefulset or Replicaset)
+        :return: Bool, string
         """
         try:
-            if pod_name:
-                log.info("Getting deploy and replicaset of pod %s", pod_name)
-                resp = self.get_deploy_replicaset(pod_name)
-                deploy = resp[1]
-            log.info("Scaling %s replicas for deployment %s", num_replica, deploy)
-            cmd = commands.KUBECTL_CREATE_REPLICA.format(num_replica, deploy)
-            output = self.execute_cmd(cmd=cmd, read_lines=True)
-            log.info("Response: %s", output)
-            time.sleep(60)
-            log.info("Check if pod of deployment %s exists", deploy)
-            cmd = commands.KUBECTL_GET_POD_DETAILS.format(deploy)
-            output = self.execute_cmd(cmd=cmd, read_lines=True, exc=False)
-            status = True if output else False
-            return status, deploy
+            set_type = None
+            if set_name:
+                set_type = const.STATEFULSET
+            elif pod_name:
+                log.info("Getting set name and set type of pod %s", pod_name)
+                set_type, set_name = self.get_set_type_name(pod_name=pod_name)
+                log.debug("Set type: %s\n Set name: %s", set_type, set_name)
```

Comment:
is this log recorded 2 time, in this function and in get_set_type_name function

Reply:
Yes, i ll keep log only in get_set_type_name() method.